docs/develop/auth.md

```diff
@@ -77,6 +76,27 @@ func AuthHandler(ctx context.Context, token string) (auth.UID, error) {
 }
 ```
 
+
+## Returning custom error messages on auth handlers
+
+When you want to return a custom error message in the auth handler e.g To return an `unauthenticated` error message:
+
+```go
+import (
+    "encore.dev/beta/auth"
+    "encore.dev/beta/errs"
+)
+
+// AuthHandler can be named whatever you prefer (but must be exported).
+//encore:authhandler
+func AuthHandler(ctx context.Context, token string) (auth.UID, error) {
+    // if the token is invalid/expired you can return a custom error message
+    // using the code below.
+    return "", errs.B().Code(errs.Unauthenticated).Msg("auth token is invalid").Err()
+}
+```
+
+
```

Comment:
Finally it would be great to add a callout on security. Something like this:

```suggestion
<Callout type="important">

Note that for security reasons you may not want to reveal too much information about why a request did not pass your auth checks. There are many subtle security considerations when dealing with authentication, and we don't have time to go into all of them here. Whenever possible we recommend using a third-party auth provider; see [Using Firebase Authentication](../how-to/firebase-auth) for an example of how to do that.

</Callout>

Reply:
@eandre i just pushed a commit to improve on the changes